archetype/src/main/resources/META-INF/archetype-post-generate.groovy

```diff
@@ -0,0 +1,142 @@
+generateMavenWrapper(outputDirectory)
+printSummary()
+
+private validateInput(jakartaVersion, javaVersion, runtime, profile, File outputDirectory){
+    if (profile.equalsIgnoreCase("core") && jakartaVersion != '10') {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, the Core Profile is only supported for Jakarta EE 10")
+    }
+
+    if (runtime.equalsIgnoreCase("payara") && (jakartaVersion != '8') && (javaVersion == '8')) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, Payara 6 does not support Java SE 8")
+    }    
+
+    if (runtime.equalsIgnoreCase("glassfish") && profile.equalsIgnoreCase("core")) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, GlassFish does not support the Core Profile")
+    }
+
+    if (runtime.equalsIgnoreCase("glassfish") && (jakartaVersion != '8') && (javaVersion == '8')) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, GlassFish 7 does not support Java SE 8")
+    }
+
+    if (runtime.equalsIgnoreCase("tomee") && (jakartaVersion == '10')) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, TomEE does not support Jakarta EE 10")
+    }
+
+    if (runtime.equalsIgnoreCase("tomee") && !profile.equalsIgnoreCase("web")) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, TomEE does not support the full and Core Profiles")
+    }
+
+    if (runtime.equalsIgnoreCase("tomee") && (jakartaVersion != '8') && (javaVersion == '8')) {
+       FileUtils.forceDelete(outputDirectory)
+       throw new RuntimeException("Failed, TomEE 9 does not support Java SE 8")
+    }    
+}
+
+private generateRuntime(runtime, jakartaVersion, docker, File outputDirectory) {
+    switch (runtime) {
+        case "glassfish": println "Generating code for GlassFish"
+            if (docker.equalsIgnoreCase("yes")) {
+                println "WARNING: GlassFish does not support Docker"
+                FileUtils.forceDelete(new File(outputDirectory, "Dockerfile"))
+            }
+        
+            break
+
+        case "tomee": println "Generating code for TomEE"
+            if (jakartaVersion != '8') {
+                println "WARNING: TomEE 9 is not yet a production ready release"
+            }
+
+            break
+
+        case "payara": println "Generating code for Payara"
+            if ((jakartaVersion != '8') && docker.equalsIgnoreCase("yes")) {
+                println "WARNING: Payara does not yet support Docker for this Jakarta EE version"
+                FileUtils.forceDelete(new File(outputDirectory, "Dockerfile"))
+            }
+            
+            break
+
+        default: println "No runtime will be included in the sample"
+    }
+}
+
+private bindEEPackage(jakartaVersion, File outputDirectory) {
+    def eePackage = 'javax';
+    if (jakartaVersion != '8') {
```

Comment:
I would turn this around. Asking positive questions is easier to understand for the reader.

Make the default jakarta ee and only change to javax when `jakartaVersion == 8`

Reply:
Sure, will change it ASAP.

Reply:
It's addressed. Feel free to mark resolved?